presets/workspace/models/vllm_model.go

```diff
@@ -0,0 +1,122 @@
+// limitations under the License.
+
+package models
+
+import (
+	_ "embed"
+	"time"
+
+	"gopkg.in/yaml.v2"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/klog/v2"
+
+	"github.com/kaito-project/kaito/pkg/model"
+	"github.com/kaito-project/kaito/pkg/utils/plugin"
+)
+
+const DefaultVLLMCommand = "python3 /workspace/vllm/inference_api.py"
+
+var (
+	//go:embed supported_models_best_effort.yaml
+	vLLMModelsYAML []byte
+)
+
+// VLLMCatalog is a struct that holds a list of supported models parsed
+// from preset/workspace/models/supported_models_best_effort.yaml. The YAML file is
+// considered the source of truth for the model metadata, and any
+// information in the YAML file should not be hardcoded in the codebase.
+type VLLMCatalog struct {
+	Models []model.VLLMModel `yaml:"models,omitempty"`
+}
+
+func init() {
+	vLLMCatalog := VLLMCatalog{}
+	utilruntime.Must(yaml.Unmarshal(vLLMModelsYAML, &vLLMCatalog))
+
+	// register all VLLM models
+	for _, m := range vLLMCatalog.Models {
+		utilruntime.Must(m.Validate())
+		plugin.KaitoModelRegister.Register(&plugin.Registration{
+			Name:     m.Name,
+			Instance: &vllmModel{model: m},
+		})
+		klog.InfoS("Registered VLLM model preset", "model", m.Name)
+	}
+}
+
+type vllmModel struct {
+	model model.VLLMModel
+}
+
+func (m *vllmModel) GetInferenceParameters() *model.PresetParam {
+	metaData := &model.Metadata{
+		Name:                 m.model.Name,
+		ModelType:            "text-generation",
+		Version:              m.model.Version,
+		Runtime:              "tfs",
+		DownloadAtRuntime:    true,
+		DownloadAuthRequired: true,
+	}
+
+	runParamsVLLM := map[string]string{
+		"trust-remote-code": "",
+	}
+	if m.model.DType != "" {
+		runParamsVLLM["dtype"] = m.model.DType
+	} else {
+		runParamsVLLM["dtype"] = "bfloat16"
+	}
+
+	if m.model.ToolCallParser != "" {
+		runParamsVLLM["tool-call-parser"] = m.model.ToolCallParser
+		runParamsVLLM["enable-auto-tool-choice"] = ""
+	}
+	if m.model.ChatTemplate != "" {
+		// todo: verify whether ChatTemplate file name is valid
+		runParamsVLLM["chat-template"] = "/workspace/chat_templates/" + m.model.ChatTemplate
+	}
+	if m.model.AllowRemoteFiles {
+		runParamsVLLM["allow-remote-files"] = ""
+	}
+
+	presetParam := &model.PresetParam{
+		Metadata:                *metaData,
+		TotalSafeTensorFileSize: m.model.ModelFileSizeGB,
+		DiskStorageRequirement:  m.model.DiskStorageRequirement,
+		BytesPerToken:           m.model.BytesPerToken,
+		ModelTokenLimit:         m.model.ModelTokenLimit,
+		RuntimeParam: model.RuntimeParam{
+			VLLM: model.VLLMParam{
+				BaseCommand:    DefaultVLLMCommand,
+				ModelName:      metaData.Name,
+				ModelRunParams: runParamsVLLM,
+			},
+		},
+		ReadinessTimeout: time.Duration(30) * time.Minute,
+	}
+
+	return presetParam
+}
+
+func (*vllmModel) GetTuningParameters() *model.PresetParam {
+	return nil
+}
+
+func (*vllmModel) SupportDistributedInference() bool {
```

Comment:
we need to consider this param in the preset generator. We can just set default to true at this now. The estimator will choose least nodes to run.